Building hello_world fails for xtensa: xt-xcc ERROR parsing -Wno-address-of-packed-member:  unknown flag

This used to work fine a few days ago, but after a `git pull` today I get the following:
```
$ west build -b intel_s1000_crb samples/hello_world
source directory: /media/psf/Home/src/zephyr/zephyr/samples/hello_world
build directory: /media/psf/Home/src/zephyr/zephyr/build (created)
BOARD: intel_s1000_crb (origin: command line)
Zephyr version: 1.14.99
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.7.3", minimum required is "3.4") 
-- Selected BOARD intel_s1000_crb
-- Found west: /usr/local/bin/west (found suitable version "0.5.7", minimum required is "0.5.6")
-- Loading /home/jh/src/zephyr/zephyr/boards/xtensa/intel_s1000_crb/intel_s1000_crb.dts as base
-- Overlaying /media/psf/Home/src/zephyr/zephyr/dts/common/common.dts
intel_s1000_crb.dts.pre.tmp:21.9-25.5: Warning (unique_unit_address_if_enabled): /cpus/cpu@0: duplicate unit-address (also used in node /cpus/core_intc@0)

warning: HAS_I2C_DW (defined at drivers/i2c/Kconfig.dw:9) has direct dependencies I2C with value n, but is currently being y-selected by the following symbols:
 - SOC_INTEL_S1000 (defined at /media/psf/Home/src/zephyr/zephyr/soc/xtensa/intel_s1000/Kconfig.soc:4), with value y, direct dependencies <choice> (value: y), and select condition <choice> (value: y)

warning: HAS_SPI_DW (defined at drivers/spi/Kconfig.dw:9) has direct dependencies SPI with value n, but is currently being y-selected by the following symbols:
 - SOC_INTEL_S1000 (defined at /media/psf/Home/src/zephyr/zephyr/soc/xtensa/intel_s1000/Kconfig.soc:4), with value y, direct dependencies <choice> (value: y), and select condition <choice> (value: y)
Parsing Kconfig tree in /media/psf/Home/src/zephyr/zephyr/Kconfig
Loading /home/jh/src/zephyr/zephyr/boards/xtensa/intel_s1000_crb/intel_s1000_crb_defconfig as base
Merging /home/jh/src/zephyr/zephyr/samples/hello_world/prj.conf
Configuration written to '/home/jh/src/zephyr/zephyr/build/zephyr/.config'
-- Cache files will be written to: /home/jh/.cache/zephyr
-- The C compiler identification is GNU 4.2.0
-- The CXX compiler identification is GNU 4.2.0
-- The ASM compiler identification is unknown
-- Found assembler: /home/jh/xtensa/XtDevTools/install/tools/RF-2016.4-linux/XtensaTools/bin/xt-xcc
<snip>
-- Configuring done
-- Generating done
-- Build files have been written to: /home/jh/src/zephyr/zephyr/build
[2/93] Preparing syscall dependency handling

[8/93] Building C object zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj
FAILED: zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj 
ccache /home/jh/xtensa/XtDevTools/install/tools/RF-2016.4-linux/XtensaTools/bin/xt-xcc -DBUILD_VERSION=zephyr-v1.14.0-1288-ga2c84ecf9e07 -DKERNEL -D_FORTIFY_SOURCE=2 -D__ZEPHYR__=1 -I../kernel/include -I/media/psf/Home/src/zephyr/zephyr/arch/xtensa/include -I../include -I../include/drivers -Izephyr/include/generated -I/media/psf/Home/src/zephyr/zephyr/soc/xtensa/intel_s1000 -I/media/psf/Home/src/zephyr/zephyr/soc/xtensa/intel_s1000/include -I../lib/libc/minimal/include -Os -g -imacros /home/jh/src/zephyr/zephyr/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -fms-extensions -D__SIZEOF_LONG__=4 -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-address-of-packed-member -Wno-main -Wno-pointer-sign -Wpointer-arith -Werror=implicit-int -fno-pic -fno-strict-overflow -ffunction-sections -fdata-sections -mlongcalls -std=c99 -MD -MT zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj -MF zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj.d -o zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj   -c /home/jh/src/zephyr/zephyr/arch/xtensa/core/offsets/offsets.c
xt-xcc ERROR parsing -Wno-address-of-packed-member:  unknown flag
ninja: build stopped: subcommand failed.
ERROR: command exited with status 1: /usr/bin/cmake --build /media/psf/Home/src/zephyr/zephyr/build
run as "west -v build -b intel_s1000_crb samples/hello_world" for a stack trace
```

--- a/cmake/compiler/gcc/target_warnings.cmake
+++ b/cmake/compiler/gcc/target_warnings.cmake
@@ -9,10 +9,15 @@ macro(toolchain_cc_warning_base)
     -Wformat
     -Wformat-security
     -Wno-format-zero-length
-    # FIXME: Remove once #16587 is fixed
-    -Wno-address-of-packed-member
     -Wno-main
   )
+
+if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "9.1.0")
+  zephyr_compile_options(
+    # FIXME: Remove once #16587 is fixed
+    -Wno-address-of-packed-member
+  )
+endif()
 
   zephyr_cc_option(-Wno-pointer-sign)
 

